propagator/opentelemetry-propagator-b3/src/opentelemetry/propagators/b3/__init__.py

```diff
@@ -46,27 +103,51 @@ class B3MultiFormat(TextMapPropagator):
     _trace_id_regex = re_compile(r"[\da-fA-F]{16}|[\da-fA-F]{32}")
     _span_id_regex = re_compile(r"[\da-fA-F]{16}")
 
+    def __init__(
+            self, *,
+            sampling_arbiter: typing.Optional[PropagatorSamplingArbiterInterface] = None):
+        """
+        NOTE: The B3 spec calls for "defer the decision to the receiver" whenever
+        explicit sampling information is absent from the carrier; this concept
+        does not translate to SpanContext, so we will invoke the Propagator
+        Sampling Arbiter as tie-breaker whenever sampling information is absent
+        from the carrier. (https://github.com/openzipkin/b3-propagation)
+
+        :param sampling_arbiter: A user-supplied arbiter instance to decide the sampled state
+            for our `extract()` method whenever sampling information is absent from carrier.
+            If None, will fall back to the legacy behavior for backward compatibility: defaults
+            decision to enabled for single-header and to disabled for multi-header (in the
+            event explicit sampling information is absent from the carrier)
+        """
+        super().__init__()
+        if sampling_arbiter is not None:
+            self._single_sampling_arbiter = self._multi_sampling_arbiter = sampling_arbiter
+        else:
+            # This reflects the original hard-coded defaults for backward compatibility
+            self._single_sampling_arbiter = default_to_accept_arbiter
+            self._multi_sampling_arbiter = default_to_deny_arbiter
+
     def extract(
         self,
         carrier: CarrierT,
         context: typing.Optional[Context] = None,
         getter: Getter = default_getter,
     ) -> Context:
+        """ Extracts SpanContext from the carrier."""
+        original_context = context
         if context is None:
             context = Context()
         trace_id = trace.INVALID_TRACE_ID
         span_id = trace.INVALID_SPAN_ID
-        sampled = "0"
+        sampled = None
         flags = None
 
         single_header = _extract_first_element(
             getter.get(carrier, self.SINGLE_HEADER_KEY)
         )
         if single_header:
-            # The b3 spec calls for the sampling state to be
-            # "deferred", which is unspecified. This concept does not
-            # translate to SpanContext, so we set it as recorded.
```

Comment:
The b3 spec calls for the sampling state to be deferred "to the receiver" whenever remote doesn't supply it. This pull request facilitates allowing the receiver to make the decision in this scenario.

See https://github.com/openzipkin/b3-propagation